### PR TITLE
Update PATCH block device documentation.

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Documentation improvement.

## Description of Changes

Provided in commit message.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Review by acatangiu — APPROVED** [review]

File: docs/api_requests/patch-block.md [hunk 1]
```diff
@@ -19,39 +26,44 @@ curl --unix-socket ${socket} -i \
...
-            \"path_on_host\": \"${ro_drive_path}\"
+             \"drive_id\": \"scratch\",            
+             \"path_on_host\": \"${updated_ro_drive_path}\"         
          }"
+         
+# It's now safe to mount the block device in the guest and use it
+# with the updated backing file.
+```
 
-# Move the backing file.
```
Comment (raduiliescu) [inline docs/api_requests/patch-block.md:45]: Don't we want to still give an example where the backing file is moved? Also looking now would be nice to offer an example where we simulate a hotplug

File: docs/api_requests/patch-block.md [hunk 2]
```diff
@@ -19,39 +26,44 @@ curl --unix-socket ${socket} -i \
      -X PUT "http://localhost/drives/scratch" \
      -H "accept: application/json" \
      -H "Content-Type: application/json" \
-     -d "{
-            \"drive_id\": \"scratch\",
-            \"path_on_host\": \"${ro_drive_path}\",
-            \"is_root_device\": false,
-            \"is_read_only\": true
+     -d "{
```
Comment (raduiliescu) [inline docs/api_requests/patch-block.md:29]: Nit: trailing whitespaces

File: docs/api_requests/patch-block.md [hunk 3]
```diff
@@ -19,39 +26,44 @@ curl --unix-socket ${socket} -i \
...
-# Even though the path has not changed, this triggers a device rescan.
+# PATCH the block device to use the new backing file.
 curl --unix-socket ${socket} -i \
      -X PATCH "http://localhost/drives/scratch" \
      -H "accept: application/json" \
      -H "Content-Type: application/json" \
      -d "{
-            \"drive_id\": \"scratch\",
-            \"path_on_host\": \"${ro_drive_path}\"
+             \"drive_id\": \"scratch\",
```
Comment (raduiliescu) [inline docs/api_requests/patch-block.md:51]: Nit: trailing white spaces.

File: docs/api_requests/patch-block.md (hunk 1 above)
Comment (sandreim) [reply docs/api_requests/patch-block.md:45]: Essentially this is unplug old block file, plug in new block file.

Comment (acatangiu) [reply docs/api_requests/patch-block.md:45]: Current example is showing how to replace backing file. This is the recommended flow so our example shows exactly that.

**Review by raduiliescu — APPROVED** [review]

**Pull request merged**